sql/tests: TestRandomSyntaxSQLSmith failed [partitions require enterprise license]

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11770371?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11770371?buildTab=artifacts#/) on master @ [0cde11b885a5e156b970c003505acd081fb4f326](https://github.com/cockroachdb/cockroach/commits/0cde11b885a5e156b970c003505acd081fb4f326):


```
=== RUN   TestRandomSyntaxSQLSmith
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSchemaChangeColumn3824830113
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith3855995623
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:124: automatically injected virtual cluster under test; see comment at top of test_server_shim.go for details.
    rsg_test.go:662: SET sql_safe_updates = false;;
    rsg_test.go:662: SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;;
    rsg_test.go:662: SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;;
    rsg_test.go:662: CREATE TABLE table1 (col1_0 TSQUERY NULL, col1_1 BIT(49) NOT NULL, col😫1_2 BOX2D, """c%pol1_3" STRING COLLATE en_US NOT NULL, " col1_4" INT4, col1_5 STRING COLLATE en_US, col1_6 BIT(26) NOT NULL, col1_7 INT8 AS (" col1_4" + 1127741478:::INT8) STORED, "\\U00010575col1_8" STRING NULL AS (lower(CAST(col1_0 AS STRING))) STORED, "!col1_9" STRING AS (lower(CAST(col1_5 AS STRING))) STORED, co😙l1_10 INT8 AS (" col1_4" + NULL) VIRTUAL, "col1_\\x7311" STRING NOT NULL AS (lower(CAST(col1_1 AS STRING))) STORED, col1_12 STRING AS (lower(CAST(col1_5 AS STRING))) VIRTUAL, UNIQUE ("""c%pol1_3" ASC, " col1_4", col1_1 ASC, col1_7) STORING (col1_0, col1_6, "!col1_9"), INDEX (col1_1 ASC, col1_5 DESC, col1_7, " col1_4" DESC) STORING (col1_0, col😫1_2, """c%pol1_3", col1_6, "col1_\\x7311") VISIBILITY 0.81, INDEX (lower(CAST(col1_5 AS STRING)) DESC, "col1_\\x7311" DESC, """c%pol1_3" DESC, lower(CAST(col1_0 AS STRING)) DESC, col1_7) STORING (col😫1_2, col1_5, "\\U00010575col1_8") WHERE (table1.co😙l1_10 != (-128):::INT8) OR (table1."!col1_9" < e'\U00002603':::STRING) NOT VISIBLE, FAMILY (col1_0), FAMILY ("col1_\\x7311"), FAMILY (" col1_4", "!col1_9"), FAMILY ("\\U00010575col1_8"), FAMILY ("""c%pol1_3"), FAMILY (col1_6), FAMILY (col1_7, col1_1), FAMILY (col1_5, col😫1_2));
    rsg_test.go:662: CREATE TABLE "tab	le2" (col2😘_0 INT8 NOT NULL, col2_1 REGROLE NOT NULL, "col2'_2" PG_LSN NOT NULL, col2_3 FLOAT8 NULL, col2_4 BIT(2) NOT NULL, col2_5 STRING NOT NULL AS (lower(CAST(col2_4 AS STRING))) STORED, PRIMARY KEY (col2_4 DESC, col2_1 ASC), UNIQUE (col2_3, col2_4, "col2'_2" ASC, col2_1 DESC) WHERE (("tab	le2".col2😘_0 = 0:::INT8) OR ("tab	le2".col2_5 >= '"':::STRING)) AND ("tab	le2".col2_3 > 1.401298464324817e-45:::FLOAT8), UNIQUE (col2_3, col2_4, col2_5 ASC, (col2_3 + (-0.6883075479578443):::FLOAT8) ASC, col2😘_0, lower(CAST(col2_4 AS STRING)) ASC) STORING ("col2'_2"), INDEX (col2_3 DESC, col2_4 DESC));
    rsg_test.go:834: pq: use of partitions requires an enterprise license. see https://cockroachlabs.com/pricing?cluster=3bdb4d86-009d-451e-3692-ca83f20c3c33 for details on how to enable enterprise features
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith3855995623
--- FAIL: TestRandomSyntaxSQLSmith (4.11s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31531

--- a/pkg/sql/tests/rsg_test.go
+++ b/pkg/sql/tests/rsg_test.go
@@ -23,6 +23,7 @@ import (
 	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/ccl"
 	"github.com/cockroachdb/cockroach/pkg/internal/rsg"
 	"github.com/cockroachdb/cockroach/pkg/internal/sqlsmith"
 	"github.com/cockroachdb/cockroach/pkg/sql"
@@ -643,6 +644,7 @@ var ignoredRegex = regexp.MustCompile(strings.Join(ignoredErrorPatterns, "|"))
 func TestRandomSyntaxSQLSmith(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+	defer ccl.TestingEnableEnterprise()() // allow usage of partitions
 
 	var smither *sqlsmith.Smither
 

